src/v/cluster/topics_frontend.cc

```diff
@@ -841,6 +843,27 @@ ss::future<topic_result> topics_frontend::do_create_topic(
     if (!units) {
         co_return make_error_result(assignable_config.cfg.tp_ns, units.error());
     }
+
+    // Enforce cluster-wide topic count limit only on user created topics.
+    auto max_user_topics_opt = _max_user_topics();
+    if (max_user_topics_opt.has_value()) {
+        // This is intended to approximate the number of topics that are
+        // user-defined and may return a count slightly smaller than the actual
+        // amount.
+        auto user_topic_count = std::max(
+                                  _topics.local().all_topics_count(),
+                                  model::non_user_topics.size())
+                                - model::non_user_topics.size();
+        if (
+          user_topic_count > *max_user_topics_opt
```

Comment:
Though maybe this math isn't exact per above, shouldn't this be >= ?

I.e., we haven't created the topic yet, so if limit is 10 and there are 10 existing topics we should fail to create the new one, but this won't fail until there are 11.